src/crimson/osd/osd_operations/background_recovery.cc

```diff
@@ -64,7 +65,12 @@ seastar::future<bool> UrgentRecovery::do_recovery()
   if (!pg->has_reset_since(epoch_started)) {
     auto futopt = pg->get_recovery_handler()->recover_missing(soid, need);
     assert(futopt);
-    return with_blocking_future(std::move(*futopt)).then([] {
+    return with_blocking_future_interruptible(std::move(*futopt))
+    .handle_interruption<
+      ::crimson::common::actingset_changed,
+      ::crimson::common::system_shutdown_exception>([](auto&& e) {
```

Comment:
Both of these are specific to osd, maybe they should be in crimson::os so that you don't have to fully qualify the namespace everywhere?